src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -46,12 +46,20 @@ public static void Verify(ImmutableDictionary<BoundExpression, (NullabilityInfo
                     {
                         if (!verifier._visitedExpressions.Contains(analyzedNode))
                         {
-                            Debug.Assert(false, $"Analyzed {verifier._analyzedNullabilityMap.Count} nodes in NullableWalker, but DebugVerifier expects {verifier._visitedExpressions.Count}. Example of unanalyzed node: {analyzedNode.GetDebuggerDisplay()}");
+                            Debug.Assert(false, $"Analyzed {verifier._analyzedNullabilityMap.Count} nodes in NullableWalker, but DebugVerifier expects {verifier._visitedExpressions.Count}. Example of unverified node: {analyzedNode.GetDebuggerDisplay()}");
+                        }
+                    }
+                }
+                else if (verifier._analyzedNullabilityMap.Count < verifier._visitedExpressions.Count)
+                {
+                    foreach (var verifiedNode in verifier._visitedExpressions)
+                    {
+                        if (!verifier._analyzedNullabilityMap.ContainsKey(verifiedNode))
+                        {
+                            Debug.Assert(false, $"Analyzed {verifier._analyzedNullabilityMap.Count} nodes in NullableWalker, but DebugVerifier expects {verifier._visitedExpressions.Count}. Example of unanalyzed node: {verifiedNode.GetDebuggerDisplay()}");
```

Comment:
The one worry I have with this form is what happens in CI. If this happens locally we can easily attach a debugger, step through and see all the nodes. In CI though we throw on the first `Debug.Assert` and hence only get the message for the first node. It's possible the first node here turns out to be a symptom, but not the root of the problem. I wonder if it's worth changing this such that the assert contains all the nodes. 

Could approach this by just creating a `StringBuilder`, and everywhere we `Debug.Assert(false, ...` now we just append the message to the builder. Then at the end of `Verify` we assert if the `StringBuilder` isn't empty. 

Curious what others thoughts are here.

Reply:
This might improve things, but I'm also happy to take the PR as-is.